test/coverlet.core.tests/Coverage/InstrumenterHelper.cs

```diff
@@ -341,8 +341,19 @@ public static CoverageResult GetCoverageResult(string filePath)
             {
                 Assert.DoesNotContain("not found for module: ", message);
             });
+
+            var serviceCollection = new ServiceCollection();
```

Comment:
We have to return same container in `GetCoverageResult` and `Run`.

They run out of process to simulate real behaviour(real instumentation), I propose to create a local static IServiceProvider that will be set inside `Run` and used in `GetCoverageResult` they're always used in sequence and so we simulate real behaviour because we keep `disableRestoreModules` container semantic.

something similar to

```cs
static void SetContainer(bool disableRestoreModules)
{
   var serviceCollection = new ServiceCollection();
            serviceCollection.AddTransient<IRetryHelper, CustomRetryHelper>();
            serviceCollection.AddTransient<IProcessExitHandler, CustomProcessExitHandler>();
            serviceCollection.AddTransient<IFileSystem, FileSystem>();
            if (disableRestoreModules)
            {
                serviceCollection.AddSingleton<IInstrumentationHelper, InstrumentationHelperForDebugging>();
            }
            else
            {
                serviceCollection.AddSingleton<IInstrumentationHelper, InstrumentationHelper>();
            }
         _processWideContainer = serviceCollection.BuildServiceProvider();
}
async public static Task<CoveragePrepareResult> Run<T>(Func<dynamic, Task> callMethod, Func<string, string[]> includeFilter = null, Func<string, string[]> excludeFilter = null, string persistPrepareResultToFile = null, bool disableRestoreModules = false)
{
   ...
  SetContainer(disableRestoreModules);
  ...
  
}
public static CoverageResult GetCoverageResult(string filePath)
{
    ...Coverage coverage = new Coverage(coveragePrepareResultLoaded, _processWideContainer.Get...,
}
```
And add this check ` Assert.DoesNotContain("not found for module: ", message);` at the end of verbose method in `Tests.Logger`.

Reply:
Ok I changed it but now some tests fail. Before the change, the static ```DependencyInjection.Current.GetService<T>()``` returned values even when ```DependencyInjection.Set(...)``` wasn't called before.